src/Cake.Npm/AddUser/AuthType.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace Cake.Npm.AddUser
+{
+    /// <summary>
+    /// Contains the valid auth strategies for <see cref="NpmAddUser"/>.
+    /// </summary>
+    public enum AuthType
```

Comment:
I think we should add an additional entry `Default` (for value 0) which means that no `--auth-type` is passed to `npm adduser` and therefore the default defined by npm is used. In case npm changes its default value this, would be the changed value compared to always defining Legacy.